Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Ride.hs

```diff
@@ -244,51 +244,65 @@ mkBookingStatus ride now = do
     DRide.CANCELLED -> Common.CANCELLED
 
 ---------------------------------------------------------------------
-rideSync :: ShortId DM.Merchant -> Id Common.Ride -> Flow Common.RideSyncRes
-rideSync merchantShortId reqRideId = do
+rideSync :: ShortId DM.Merchant -> Common.RideSyncReq -> Flow Common.RideSyncRes
+rideSync merchantShortId rideSyncReq = do
   merchant <- findMerchantByShortId merchantShortId
-  let rideId = cast @Common.Ride @DRide.Ride reqRideId
-  ride <- runInReplica $ QRide.findById rideId >>= fromMaybeM (RideDoesNotExist rideId.getId)
-  booking <- runInReplica $ QBooking.findById ride.bookingId >>= fromMaybeM (BookingNotFound rideId.getId)
+  let rideIds = map (cast @Common.Ride @DRide.Ride) rideSyncReq.rideIds
+  rides <- mapM (\rideId -> runInReplica $ QRide.findById rideId >>= fromMaybeM (RideDoesNotExist rideId.getId)) rideIds
+  let rideIds_rides_zip = zip rideIds rides
+  bookings <- mapM (\(rideId, ride) -> runInReplica $ QBooking.findById ride.bookingId >>= fromMaybeM (BookingNotFound rideId.getId)) rideIds_rides_zip
 
+  let bookings_rideIds_zip = zip bookings rideIds
   -- merchant access checking
-  unless (merchant.id == booking.providerId) $ throwError (RideDoesNotExist rideId.getId)
-
-  logTagInfo "dashboard -> syncRide : " $ show rideId <> "; status: " <> show ride.status
+  for_ bookings_rideIds_zip $ \(booking, rideId) -> do unless (merchant.id == booking.providerId) $ throwError (RideDoesNotExist rideId.getId)
+  logTagInfo "dashboard -> syncRide : " $ show rideIds <> "; status: " <> show (map (\bookings_rideIds_zip -> bookings_rideIds_zip.status) rides)
```

Comment:
can be refactored using dot record syntax:

`logTagInfo "dashboard -> syncRide : " $ show rideIds <> "; status: " <> show (rides <&> (.status))`